File: output/gpt_4o/avatar/algo_based_translation/python/java/successful_translation/codeforces_705_B.java

import java.util.Scanner;

public class codeforces_705_B {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        String[] input = scanner.nextLine().split(" ");
        int[] w = new int[n];
        for (int j = 0; j < n; j++) {
            w[j] = Integer.parseInt(input[j]);
        }
        int c = 2;

        for (int i : w) {
            if (i == 1) {
                System.out.println(c);
            } else if (i % 2 == 0) {
                c = 3 - c;
                System.out.println(c);
            } else {
                System.out.println(c);
            }
        }
        scanner.close();
    }
}

File: output/gpt_4o/avatar/algo_based_translation/java/python/successful_translation/codeforces_705_B.py

import sys

class FastReader:
    def __init__(self):
        self.input = sys.stdin.read
        self.data = self.input().split()
        self.index = 0

    def next(self):
        if self.index < len(self.data):
            result = self.data[self.index]
            self.index += 1
            return result
        return None

    def nextInt(self):
        return int(self.next())

    def nextLong(self):
        return int(self.next())

    def nextDouble(self):
        return float(self.next())

    def nextLine(self):
        return self.next()

def solve():
    t = scan.nextInt()
    arr = [0] * t
    for i in range(t):
        arr[i] = scan.nextInt()

    prevWinner = 0

    for i in range(t):
        if arr[i] == 1:
            if prevWinner == 0:
                prevWinner = 2

        if prevWinner == 2 or prevWinner == 0:
            if (arr[i] - 1) % 2 == 0:
                print(2)
                prevWinner = 2
            else:
                print(1)
                prevWinner = 1
        else:
            if (arr[i] - 1) % 2 == 0:
                print(1)
                prevWinner = 1
            else:
                print(2)
                prevWinner = 2

if __name__ == "__main__":
    scan = FastReader()
    t = 1
    while t > 0:
        t -= 1
        solve()

